### PR TITLE
Batched dispatching

### PR DESCRIPTION
**Pull request by SzymonPobiega**
Changed the API of `IDispatchMessages` to be able to pass a collection of `TransportOperation`s to be dispatched to the transport. Rationale:
 * It allows for batching sends which can improve performance of certain transports
 * It allows to get rid of `StorageDrivenDispatcher`
 * It makes the concept of the transport operation explicit

### CODE REVIEW

**Comment by SzymonPobiega** [comment]
@andreasohlund @danielmarbach please take a look

**Comment by danielmarbach** [comment]
The name clash between outbox and transport is not so cool. Is there a way around it?

**Comment by SzymonPobiega** [comment]
@danielmarbach I believe the name of that outbox thing should have `DTO` suffix but I am afraid that might break downstreams (persistences) in subtle ways if they don't use their own classes when storing.

File: src/NServiceBus.Core/Transports/TransportOperation.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
+namespace NServiceBus.Transports
+{
+    /// <summary>
+    /// Defines the transport operations including the message and information how to send it.
+    /// </summary>
+    public class TransportOperation
+    {
+        OutgoingMessage message;
+        DispatchOptions dispatchOptions;
```
Comment (andreasohlund) [inline src/NServiceBus.Core/Transports/TransportOperation.cs]: Using properties with private setters seems to be our standard and is less lines of code?

**Comment by SzymonPobiega** [comment]
@andreasohlund rebased and fixed the property. Can't build it locally because my battery is dying so please wait with the merge till the build server confirms I haven't broken anything.

**Pull request merged**